src/local/butler/scripts/migration/monorail_to_issuetracker_migrator.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Migrate Issue Ids from Monorail to Issue Tracker."""
+
+import csv
+
+from google.cloud import ndb
+
+from clusterfuzz._internal.datastore import data_types
+
+
+def execute(args):
+  """Query Testcases of a project, and update the bug_information
+  and/or group_bug_information fields to reflect the Issue Tracker issue
+  id rather than the Monorail issue id."""
+
+  issue_id_dict = get_monorail_issuetracker_issue_id_dictionary()
+
+  testcases = []
+
+  for testcase in data_types.Testcase.query(
+      # only target testcases in single project
+      data_types.Testcase.project_name == args.project_name,):
+    if testcase.bug_information and issue_id_dict.get(testcase.bug_information):
+      testcase.bug_information = ndb.StringProperty(
+          issue_id_dict[testcase.bug_information])
+
+    if testcase.group_bug_information and issue_id_dict.get(testcase.group_bug_information):
+      testcase.group_bug_information = ndb.IntegerProperty(
+          issue_id_dict[testcase.group_bug_information])
+
+    testcases.append(testcase)
+
+    if args.non_dry_run and len(testcases) > 499: # maximum batch size is 500
```

Comment:
Since our maximum batch size is 500, I'm making sure we run a batch job every 500 testcases processed, and then emptying the queue to allow the process to continue. We do a final check for the remaining testcases at the end.

Let me know if you see a more elegant way to handle this.

Reply:
I think this means that the last batch of less than 500 will not be updated.
Maybe better to finish populating testcases, and then outside the loop to process chunks of it.

Reply:
Nevermind. As discussed, I misread line:51.

Reply:
Discussed this over chat - this concern is addressed by the final ndb.put_multi on line 51.